zipline/algorithm.py

```diff
@@ -883,7 +883,12 @@ def get_datetime(self, tz=None):
         """
         Returns the simulation datetime.
         """
-        dt = self.datetime
+        # If datetime is not yet set, use the period start.
+        if self.datetime is not None:
```

Comment:
Can we just set self.datetime to period_start at construction time so that this is never None?